cmd/influxd/launcher/launcher_test.go

```diff
@@ -164,3 +168,21 @@ func TestLauncher_PingHeaders(t *testing.T) {
 	assert.Equal(t, []string{"OSS"}, resp.Header.Values("X-Influxdb-Build"))
 	assert.Equal(t, []string{"dev"}, resp.Header.Values("X-Influxdb-Version"))
 }
+
+func TestLauncher_PIDFile(t *testing.T) {
+	pidDir := t.TempDir()
+	pidFilename := filepath.Join(pidDir, "influxd.pid")
+
+	l := launcher.RunAndSetupNewLauncherOrFail(ctx, t, func(o *launcher.InfluxdOpts) {
+		o.PIDFile = pidFilename
+	})
+	defer func() {
+		l.ShutdownOrFail(t, ctx)
```

Comment:
It's inside a lambda, so it will not be evaluated until the defer calls the lambda. Also note that if it got called immediately the `require.FileExists()` on line 184 would fail because the PID file gets removed when `l.ShutdownOrFail(t, ctx)` is called.

Reply:
@devanbenz - the arguments to a deferred function are evaluated on the execution of the `defer` line, and those evaluated arguments are stored for the eventual call, but the deferred call itself doesn't happen until after the end of the closing method.